crates/holochain/src/core/workflow/publish_dht_ops_workflow.rs

```diff
@@ -57,6 +65,92 @@ pub async fn publish_dht_ops_workflow(
     let mut success = Vec::with_capacity(to_publish.len());
     for (basis, list) in to_publish {
         let (op_hash_list, op_data_list): (Vec<_>, Vec<_>) = list.into_iter().unzip();
+
+        let mut have_enough_receipts = false;
+
+        // First, check to see if we can get the required validation receipts.
+        // TODO - we don't currently track the agents that generated the
+        //        receipts in our database, so we can't make use of the
+        //        exclude list, so we may get duplicate receipts.
+        if let Ok(bundle) = network
+            .get_validation_receipts(
+                basis.clone(),
+                op_hash_list.clone(),
+                Vec::new(), // exclude list
+                DEFAULT_RECEIPT_BUNDLE_SIZE as usize,
+            )
+            .await
+        {
+            for receipt in bundle.into_iter() {
+                // Get the action for this op so we can check the entry type.
+                let hash = receipt.receipt.dht_op_hash.clone();
+                let action: Option<SignedAction> = authored_db
+                    .read_async(move |txn| {
+                        let h: Option<Vec<u8>> = txn
+                            .query_row(
+                                "SELECT Action.blob as action_blob
+                    FROM DhtOp
+                    JOIN Action ON Action.hash = DhtOp.action_hash
+                    WHERE DhtOp.hash = :hash",
+                                named_params! {
+                                    ":hash": hash,
+                                },
+                                |row| row.get("action_blob"),
+                            )
+                            .optional()?;
+                        match h {
+                            Some(h) => from_blob(h),
+                            None => Ok(None),
+                        }
+                    })
+                    .await?;
```

Comment:
Database errors terminate the workflow run. Should this be contained to the current batch of ops?

Reply:
Done: https://github.com/holochain/holochain/pull/4796/commits/70e08c131e099d1e918ab4dd5f3b884885ba7e4e